### PR TITLE
WIP: Separate modbus_adapter_sto and stop1_executor

### PR DESCRIPTION
**Pull request by hslusarek**

### CODE REVIEW

File: pilz_testutils/include/pilz_testutils/async_test.h [hunk 1]
```diff
@@ -98,6 +98,7 @@ class AsyncTest
 // for better readability in tests
 #define BARRIER(...) barricade(__VA_ARGS__)
 #define BARRIER2(str1, str2) barricade(str1, str2)
+#define BARRIER3(str1, str2, str3) barricade(str1, str2, str3)
```
Comment (agutenkunst) [inline pilz_testutils/include/pilz_testutils/async_test.h]: Please use BARRIER(...)

File: prbt_hardware_support/include/prbt_hardware_support/modbus_adapter_sto.h [hunk 2]
```diff
@@ -19,27 +19,28 @@
...
 
 #include <prbt_hardware_support/ModbusMsgInStamped.h>
-#include <prbt_hardware_support/filter_pipeline.h>
 #include <prbt_hardware_support/modbus_api_spec.h>
-#include <prbt_hardware_support/adapter_sto.h>
 
 namespace prbt_hardware_support
 {
 
+using TUpdateStoFunc = std::function<void(const bool)>;
```
Comment (agutenkunst) [inline prbt_hardware_support/include/prbt_hardware_support/modbus_adapter_sto.h]: Why the T?

File: prbt_hardware_support/include/prbt_hardware_support/stop1_executor.h [hunk 3]
```diff
@@ -0,0 +1,145 @@
+/*
```
Comment (agutenkunst) [inline prbt_hardware_support/include/prbt_hardware_support/stop1_executor.h:1]: Did you rename with "git mv"?

File: prbt_hardware_support/test/unittests/unittest_stop1_executor.cpp [hunk 4]
```diff
@@ -819,31 +809,31 @@ TEST_F(AdapterStoTest, testExitInStateStopRequestedDuringEnable)
...
-  std::function<bool()> sto_false_during_recover_action = [this, &adapter_sto]() {
+  TServiceCallFunc sto_false_during_recover_action = [this, &adapter_sto]() {
     this->triggerClearEvent(RECOVER_SRV_CALLED_EVENT);
-    adapter_sto.updateSto(false);
+    adapter_sto->updateSto(false);
     return true;
   };
 
-  std::function<bool()> false_recover_action = []() {
+  TServiceCallFunc false_recover_action = []() {
```
Comment (agutenkunst) [inline prbt_hardware_support/test/unittests/unittest_stop1_executor.cpp]: Would auto work?

File: prbt_hardware_support/test/unittests/unittest_stop1_executor.cpp [hunk 5]
```diff
@@ -819,31 +809,31 @@ TEST_F(AdapterStoTest, testExitInStateStopRequestedDuringEnable)
...
     return false;
   };
 
-  std::function<bool()> enable_during_halt_action = [this, &adapter_sto]() {
-    adapter_sto.updateSto(true);
-    adapter_sto.stopStateMachine();
+  TServiceCallFunc enable_during_halt_action = [this, &adapter_sto]() {
```
Comment (agutenkunst) [inline prbt_hardware_support/test/unittests/unittest_stop1_executor.cpp]: Would auto work

**Review by agutenkunst — CHANGES_REQUESTED** [review]
Just some minor stuff.

File: pilz_testutils/include/pilz_testutils/async_test.h (hunk 1 above)
Comment (hslusarek) [reply pilz_testutils/include/pilz_testutils/async_test.h]: Done.

File: prbt_hardware_support/include/prbt_hardware_support/modbus_adapter_sto.h (hunk 2 above)
Comment (hslusarek) [reply prbt_hardware_support/include/prbt_hardware_support/modbus_adapter_sto.h]: Done.

File: prbt_hardware_support/include/prbt_hardware_support/stop1_executor.h (hunk 3 above)
Comment (hslusarek) [reply prbt_hardware_support/include/prbt_hardware_support/stop1_executor.h:1]: No.

File: prbt_hardware_support/test/unittests/unittest_stop1_executor.cpp (hunk 5 above)
Comment (hslusarek) [reply prbt_hardware_support/test/unittests/unittest_stop1_executor.cpp]: Done.

File: prbt_hardware_support/test/unittests/unittest_stop1_executor.cpp (hunk 4 above)
Comment (hslusarek) [reply prbt_hardware_support/test/unittests/unittest_stop1_executor.cpp]: Done.

**Pull request merged**